File: app/src/main/java/de/bikebean/app/db/settings/settings/Battery.java
package de.bikebean.app.db.settings.settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.settings.Setting;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.sms.SmsFactory;
import de.bikebean.app.db.state.State;
import de.bikebean.app.db.state.StateFactory;

public class Battery extends Setting {

    public static final double UNSET_BATTERY = -1.0;
    private static final @NonNull State.KEY key = State.KEY.BATTERY;

    public Battery(final @NonNull Sms sms, final @NonNull BatteryGetter batteryGetter) {
        super(
                StateFactory.createNumberState(
                        sms, key, batteryGetter.getBattery(), State.STATUS.CONFIRMED
                ),
                Setting::addToList
        );
    }

    public Battery() {
        super(
                StateFactory.createNumberState(SmsFactory.createNullSms(), key, UNSET_BATTERY, State.STATUS.UNSET),
                Setting::addToList
        );
    }

    public interface BatteryGetter {
        double getBattery();
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/add_to_list_settings/Battery.kt
package de.bikebean.app.db.settings.settings.add_to_list_settings

import de.bikebean.app.db.settings.settings.AddToListSetting
import de.bikebean.app.db.sms.Sms
import de.bikebean.app.db.state.State
import de.bikebean.app.db.state.State.KEY
import de.bikebean.app.db.state.StateFactory

class Battery : AddToListSetting {

    constructor(
            sms: Sms, batteryGetter: () -> Double
    ) : super(StateFactory.createNumberState(
            sms, key, batteryGetter(), State.STATUS.CONFIRMED
    ))

    constructor() : super(StateFactory.createUnsetState(key, UNSET_BATTERY))

    companion object {
        const val UNSET_BATTERY = -1.0
        private val key = KEY.BATTERY
    }
}